src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -93,17 +123,21 @@ def remove_vm_from_nsgroups(vm)
       lports = logical_ports(vm)
 
       lport_ids = lports.map(&:id)
-      nsgroups = @client.nsgroups.select do |nsgroup|
+      nsgroups = grouping_obj_svc.list_ns_groups.results.select do |nsgroup|
         not nsgroup.members.select do |member|
```

Comment:
Wow I didn't realize this was in the codebase. Instead of `not ... select ... empty?` should just be an `any?`.

Reply:
fixed